crates/token/src/lib.rs

```diff
@@ -108,21 +101,114 @@ impl Transfer {
     BorshDeserialize,
     BorshDeserializer,
     BorshSchema,
+    Default,
     Hash,
     Eq,
     PartialOrd,
     Serialize,
     Deserialize,
 )]
-pub struct TransferData {
-    /// Source address will spend the tokens
-    pub source: Address,
-    /// Target address will receive the tokens
-    pub target: Address,
-    /// Token's address
-    pub token: Address,
-    /// The amount of tokens
-    pub amount: DenominatedAmount,
+pub struct Transfer {
+    /// Sources of this transfer
+    pub sources: BTreeMap<Account, DenominatedAmount>,
```

Comment:
Question, do we lose information about the specific transfers with this struct? @sug0 has placed some`TokenEvent`s around the codebase and I think we don't have all the data to construct those events anymore (specifically we don't have the exact mapping between source and target)

Reply:
Good point, Yes we lose specific information. @sug0 Could we create a multi-transfer event that doesn't require exact mappings between sources and targets?

Reply:
let's change the `TokenOperation::Transfer` (in `crates/trans_token/src/event.rs`) to support this

Reply:
> let's change the `TokenOperation::Transfer` (in `crates/trans_token/src/event.rs`) to support this

I've attempted to do this in e4da8c0433ec323310d64dc8f4e106e157a9b0f1 . Would you be able to review again?

Reply:
the change lgtm, but some tests seem to be affected

Reply:
@murisi it looks good to me but tests are failing

Reply:
Ah, fixed. Thanks!